Library/Homebrew/livecheck/strategy/crates.rb

```diff
@@ -0,0 +1,111 @@
+# typed: true
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    module Strategy
+      # The {Crates} strategy identifies versions of a crates.io package by
+      # checking the information from the `versions` API endpoint.
+      #
+      # Crate URLs have the following format:
+      #   `https://static.crates.io/crates/example/example-1.2.3.crate`
+      #
+      # The default regex identifies versions like `1.2.3`/`v1.2.3` from the
+      # version `num` field. This is a common version format but a different
+      # regex can be provided in a `livecheck` block to override the default
+      # if a package uses a different format (e.g. `1.2.3d`, `1.2.3-4`, etc.).
+      #
+      # @api public
+      class Crates
+        # The default regex used to identify versions when a regex isn't
+        # provided.
+        DEFAULT_REGEX = /^v?(\d+(?:\.\d+)+)$/i
+
+        # The default `strategy` block used to extract version information when
+        # a `strategy` block isn't provided.
+        DEFAULT_BLOCK = proc do |json, regex|
+          json["versions"]&.map do |version|
+            next if version["yanked"] == true
+            next unless (match = version["num"]&.match(regex))
+
+            match[1]
```

Comment:
```suggestion
            version["num"]&.slice(regex, 1)
```

Reply:
While the `#slice` approach is tidier when we're only working with one capture group, the `next unless match...` pattern is what we use in `strategy` blocks throughout core/cask.

The general idea is that in the `#match` approach, we only execute the regex once regardless of the number of capture groups. Unless Ruby does some magic behind the scenes, the `#slice` approach would seemingly execute the regex each time we fetch a capture group.

We're only using one capture group in the default strategy regex, so the `#slice` approach would work but I favor the existing `#match` approach because it's consistent with what we do in `strategy` blocks.